public/app/features/alerting/unified/hooks/useAlertmanagerConfig.ts

```diff
@@ -1,26 +1,28 @@
+import { SerializedError } from '@reduxjs/toolkit';
 import { useEffect } from 'react';
 
-import { useDispatch } from 'app/types';
+import { alertmanagerApi } from '../api/alertmanagerApi';
 
-import { fetchAlertManagerConfigAction } from '../state/actions';
-import { initialAsyncRequestState } from '../utils/redux';
+type Options = {
+  refetchOnFocus?: boolean;
+  refetchOnReconnect?: boolean;
+};
 
-import { useUnifiedAlertingSelector } from './useUnifiedAlertingSelector';
-
-export function useAlertmanagerConfig(amSourceName?: string) {
-  const dispatch = useDispatch();
+// TODO refactor this so we can just call "alertmanagerApi.endpoints.getAlertmanagerConfiguration" everywhere
+// and remove this hook since it adds little value
+export function useAlertmanagerConfig(amSourceName?: string, options?: Options) {
+  const [fetchConfig, fetchState] = alertmanagerApi.endpoints.getAlertmanagerConfiguration.useLazyQuery(options);
```

Comment:
Wouldn't it be simpler to use `useGetAlertmanagerConfiguration` with the `skip: !amSourceName` option?
If not, it should be possible to use `alertmanagerApi.useLazyGetAlertmanagerConfiguration` without going through the `endpoints` object

Reply:
I think it makes sense here to use `skip: !amSourceName` here – so I'll refactor to use that instead 👍 

Regarding the endpoints object, I prefer this approach as it allows my editor to jump directly to the definition in `alertmanagerApi` and it didn't work when using the `useEndpointQuery()` style :(